pkg/pgmodel/end_to_end_tests/query_integration_test.go

```diff
@@ -38,6 +68,60 @@ func NewPromClient(urlStr string, timeout time.Duration) (*PromClient, error) {
 	return &PromClient{url: url, httpClient: httpClient}, nil
 }
 
+func (c *PromClient) genInstantRequest(query string, start time.Time) (*http.Request, error) {
```

Comment:
nit: a bit weird to have this function be part of PromClient since the returned request is also used to send to the connector. Not a big deal but may be cleaner as a standalone function